tests/unit_tests/commands/dataset/test_update.py

```diff
@@ -0,0 +1,43 @@
+from unittest.mock import MagicMock
```

Comment:
I was _blown away_ that there were [zero unit or integration tests](https://github.com/search?q=repo%3Aapache%2Fsuperset+UpdateDatasetCommand+path%3A%2F%5Etests%5C%2F%2F&type=code) for the `UpdateDatasetCommand`.

Though this test isn't pretty, it's a start.